index.php

```diff
@@ -1,17 +1,22 @@
 <?php
 
-    /**
-     * @author Max Goriunov
-     */
+/**
+ * @author Max Goriunov
+ */
 
-    use Components\Autoloader;
-    use Components\Router;
+use Components\Autoloader;
+use Routing\BsRouter;
+use Routing\NonEmptyRouter;
+use Routing\Router;
 
-    /**
-     * @todo Add working email address
-     */
-    include_once("./components/Autoloader.php");
-    include_once("./config/constants.php");
+/**
+ * @todo Add working email address
+ */
+include_once("./components/Autoloader.php");
+include_once("./config/constants.php");
 
-    (new Autoloader())->register();
-    (new Router())->run();
\ No newline at end of file
+(new Autoloader())->register();
+(new NonEmptyRouter(
+    new BsRouter()
+))
+    ->run($_SERVER["REQUEST_URI"]);
```

Comment:
Added a protective decorator for router in index.php so that nobody can invoke it with an empty request uri.